Optionally load and use Unicode::UTF8

xsawyerx opened the pull request:
Data needed. :)

SysPete commented on lib/Dancer2/Core/Request.pm:
```diff
@@ -37,6 +37,16 @@ sub $_ { \$_[0]->env->{ 'HTTP_' . ( uc "$_" ) } }
 
+eval {
+    require_module('Unicode::UTF8');
+    1;
```
what is this doing here?

xsawyerx replied:
Resolved.

xsawyerx commented:
I've rebased this PR and fixed up what was necessary.

My benchmarks showed a minimal improvement, but hey, maybe it's enough or maybe my benchmark is not accurate enough.

cromedome commented:
You might wish to add `Unicode::UTF8` as recommended in `cpanfile` (also, in `share/skel/cpanfile`).

I will comment more fully soon!

cromedome commented:
Merged with one small change to `share/skel/cpanfile`. Thanks!
